### ISSUE TITLE
⚡Request for FLUX-based LoRA Training Configuration (.toml) Example

### ISSUE THREAD

**Issue by sashaok123** [derrian-distro/LoRA_Easy_Training_Scripts#248]
Hello!

I've recently started working with the new training tab for **FLUX** in LoRA Easy Training Scripts, and I'm finding it quite challenging to configure everything properly due to the numerous new settings. The lack of intuitive guidance makes it difficult to understand the exact setup required for optimal training.

Could you kindly provide an example of a **.toml** configuration file specifically for FLUX training? Having a ready-to-use config would greatly simplify the process. Here's what I'd appreciate in the config:
- Full training settings for **FLUX.1-dev** models.
- Default or recommended parameters for a smooth start.
- Instructions on how to adjust the config for different VRAM capacities (e.g., 12 GB or more).

Your help in providing this file would be invaluable and save a lot of time for many users struggling with the new setup.

Thank you in advance for your support!

**Comment by PheonixAi420** [comment]
This will work. Just change directories and paths to your specific paths on your PC ae.saftensors, t5xxl, clip l, etc. 
[Example Toml.txt](https://github.com/user-attachments/files/17416337/Example.Toml.txt)


[[subsets]]
caption_extension = ".txt"
image_dir = "C:/StableDiffusion/training/img/10_Aurolka"
keep_tokens = 1
name = "subset 1"
num_repeats = 10
shuffle_caption = true

[train_mode]
train_mode = "lora"

[general_args.args]
max_data_loader_n_workers = 1
persistent_data_loader_workers = true
pretrained_model_name_or_path = "C:/StableDiffusion/stable-diffusion-webui/extensions/stable-diffusion-webui-forge/models/Stable-diffusion/flux1-dev.safetensors"
vae = "C:/StableDiffusion/stable-diffusion-webui/extensions/stable-diffusion-webui-forge/models/text_encoder/t5xxl_fp16.safetensors"
clip_skip = 1
no_half_vae = true
highvram = true
full_bf16 = true
mixed_precision = "bf16"
fp8_base = true
gradient_checkpointing = true
gradient_accumulation_steps = 1
seed = 23
max_token_length = 225
prior_loss_weight = 1.0
sdpa = true
max_train_epochs = 4
cache_latents = true
cache_latents_to_disk = true

[general_args.dataset_args]
resolution = [ 1024, 1024,]
batch_size = 1

[network_args.args]
network_dim = 32
network_alpha = 16.0
min_timestep = 0
max_timestep = 1000

[optimizer_args.args]
optimizer_type = "AdamW8bit"
lr_scheduler = "constant"
loss_type = "l2"
learning_rate = 0.0003
unet_lr = 0.0003
text_encoder_lr = 0.0003
max_grad_norm = 1.0
min_snr_gamma = 5

[saving_args.args]
output_dir = "C:/StableDiffusion/training/Model"
save_precision = "bf16"
save_model_as = "safetensors"
save_every_n_epochs = 1
save_toml = true
save_toml_location = "C:/StableDiffusion/training/Log"
save_state = true
output_name = "Aurolka"

[noise_args.args]
noise_offset = 0.4

[sample_args.args]
sample_sampler = "euler_a"
sample_every_n_epochs = 1
sample_prompts = "C:/StableDiffusion/training/img/002-.txt"

[flux_args.args]
ae = "C:/StableDiffusion/stable-diffusion-webui/extensions/stable-diffusion-webui-forge/models/VAE/ae.safetensors"
clip_l = "C:/StableDiffusion/stable-diffusion-webui/extensions/stable-diffusion-webui-forge/models/text_encoder/clip_l.safetensors"
t5xxl = "C:/StableDiffusion/stable-diffusion-webui/extensions/stable-diffusion-webui-forge/models/text_encoder/t5xxl_fp16.safetensors"
apply_t5_attn_mask = true
t5xxl_max_token_length = 512
split_mode = true
timestep_sampling = "sigmoid"
sigmoid_scale = 1.0
discrete_flow_shift = false
weighting_scheme = "none"
guidance_scale = 1.0
model_prediction_type = "raw"

[bucket_args.dataset_args]
enable_bucket = true
bucket_no_upscale = true
min_bucket_reso = 256
max_bucket_reso = 1024
bucket_reso_steps = 64

[network_args.args.network_args]
train_blocks = "single"

[optimizer_args.args.optimizer_args]
weight_decay = "0.1"

**Comment by sashaok123** [comment]
> This will work. Just change directories and paths to your specific paths on your PC ae.saftensors, t5xxl, clip l, etc. [Example Toml.txt](https://github.com/user-attachments/files/17416337/Example.Toml.txt)


Thanks for the provided config.

With what amount of memory do you train on such settings?
I'm missing 24Gb of VRAM on the RTX4090

**Comment by PheonixAi420** [comment]
Im running a 4060TI with 16gb vram and 32 gb ram. If you are running higher vram you can still use these setting and  it will train faster and if you are running less than 16gb vram I would recommend using the low vram option. I have trained a lot of Flux loras and I can train celebrity loras with 15-20 high quality images at about 1 epoch every 30-45 minutes and typically only need 2-3 epochs with 150-160 steps per epoch.  Below is an example of the Madelyn Cline lora I made. 
![00008-2963787977](https://github.com/user-attachments/assets/960bb0e6-f896-4d43-b5f4-71059547addf)

**Comment by Jelosus2** [comment]
Seems to be solved (?)

**Comment by AdAstra-93** [comment]
> Im running a 4060TI with 16gb vram and 32 gb ram. If you are running higher vram you can still use these setting and it will train faster and if you are running less than 16gb vram I would recommend using the low vram option. I have trained a lot of Flux loras and I can train celebrity loras with 15-20 high quality images at about 1 epoch every 30-45 minutes and typically only need 2-3 epochs with 150-160 steps per epoch. Below is an example of the Madelyn Cline lora I made. ![00008-2963787977](https://github.com/user-attachments/assets/960bb0e6-f896-4d43-b5f4-71059547addf)

@PheonixAi420 Did you use vanilla Flux to make that image, or some fine-tune? it looks pretty good actually! what resolution do you get your dataset pictures on to get that quality?
